scheduler/pkg/server/server_status.go

```diff
@@ -206,7 +225,18 @@ func (s *SchedulerServer) updateServerModelsStatus(evt coordinator.ModelEventMsg
 	}
 	s.serverEventStream.pendingLock.Unlock()
 
-	return nil
+	return err
+}
+
+func (s *SchedulerServer) incrementExpectedReplicas(server *store.ServerSnapshot) {
+	// TODO: should there be some sort of velocity check ?
+	logger := s.logger.WithField("func", "incrementExpectedReplicas")
+	logger.Debugf("will attempt to scale servers to %d for %v", server.Stats.MaxNumReplicaHostedModels, server.Name)
+
+	ssr := createServerStatusUpdateResponse(server)
+	ssr.ExpectedReplicas = int32(server.Stats.MaxNumReplicaHostedModels)
```

Comment:
should we respect `maxReplicas` on the server if defined? i.e. it should be `min(maxReplicas, server.Stats.MaxNumReplicaHostedModels)`?

Reply:
Yeah, it should - I just checked `isValidScalingEvent` in the controller.